standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/MetaStoreDirectSql.java

```diff
@@ -870,44 +870,88 @@ private boolean isViewTable(String catName, String dbName, String tblName) throw
-    long queryTime = doTrace ? System.nanoTime() : 0;
-    MetastoreDirectSqlUtils.timingTrace(doTrace, queryText, start, queryTime);
-    if (sqlResult.isEmpty()) {
-      return Collections.emptyList(); // no partitions, bail early.
+    int parametersToProcess = paramsForFilter.size();
+
+    List<Long> result = new ArrayList<>();
+
+    // If this method has been called by default we need to query the tables once. Hence set the minimum numbner of
+    // iterations to 1.
+    int iterations = 1;
+
+
+    // The first three parameters in the query are the table, database and the catalog names. This leaves us
+    // JDBC_STMT_PARAM_LIMIT - 3 parameters that can be passed to the underlying JDBC driver. If the number of
+    // input parameters is greater than this limit, we will have to process them in multiple iterations.
+    if (parametersToProcess > (JDBC_STMT_PARAM_LIMIT - 3)) {
+      // Find the number of iterations required from the number of input parameters and sets of 32764 parameters
+      // that can be created from it.
+      iterations = parametersToProcess / (JDBC_STMT_PARAM_LIMIT - 3);
+      iterations += ((parametersToProcess % (JDBC_STMT_PARAM_LIMIT - 3) != 0) ? 1 : 0);
     }
 
-    List<Long> result = new ArrayList<>(sqlResult.size());
-    for (Object fields : sqlResult) {
-      result.add(MetastoreDirectSqlUtils.extractSqlLong(fields));
+    for (int i = 0 ; i < iterations; i++) {
+      // The number of filter parameters that can be processed in each iteration has to be lesser than or equal to
+      // JDBC_STMT_PARAM_LIMIT - 3
+      int filterParamSize = parametersToProcess > (JDBC_STMT_PARAM_LIMIT - 3) ? (JDBC_STMT_PARAM_LIMIT - 3) : parametersToProcess;
+
+      Object[] params = new Object[filterParamSize + 3];
+      params[0] = tblNameLcase;
+      params[1] = dbNameLcase;
+      params[2] = catNameLcase;
+
+      // Reduce by the number of parameters that have already been processed.
+      parametersToProcess -= (JDBC_STMT_PARAM_LIMIT - 3);
+
+      for (int j = 0; j < filterParamSize; j++) {
+        // The current set of parameters have to be fetched from the correct offset in the input parameters set.
+        params[j + 3] = paramsForFilter.get(((JDBC_STMT_PARAM_LIMIT - 3) * i) + j);
+      }
+
+      long start = doTrace ? System.nanoTime() : 0;
+      Query query = pm.newQuery("javax.jdo.query.SQL", queryText);
+      List<Object> sqlResult = MetastoreDirectSqlUtils.executeWithArray(query, params, queryText, ((max == null) ? -1 : max.intValue()));
+      long queryTime = doTrace ? System.nanoTime() : 0;
+      MetastoreDirectSqlUtils.timingTrace(doTrace, queryText, start, queryTime);
+
+      for (Object fields : sqlResult) {
+        if (max != null && max != -1) {
+          if (max > 0) {
+            max--;
+          } else {
+            break;
+          }
+        }
+        result.add(MetastoreDirectSqlUtils.extractSqlLong(fields));
+      }
+
```

Comment:
remove blank line

Reply:
Will do ! Will make the change and resubmit.